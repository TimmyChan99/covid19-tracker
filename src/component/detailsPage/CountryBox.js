import React from 'react';
import { useDispatch } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { storeSelectedCountry } from '../../redux/dates/date';

const CountryBox = ({ countryName, countryNewCases }) => {
  const dispatch = useDispatch();
  const getCountryName = (e) => {
    const { id } = e.currentTarget;
    dispatch(storeSelectedCountry(id));
  };

  return (
    <div className="country d-flex-column">
      <NavLink to={`/covid19-tracker/countries-list/${countryName}/regions/`} className="arrow">
        <BsArrowRightCircle id={countryName} onClick={getCountryName} />
      </NavLink>
      <div className="info d-flex-column">
        <span className="country_name">{countryName}</span>
        <span className="stat_number">
          {countryNewCases}
          {' '}
          cases
        </span>
      </div>
      <img
        src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/ad/vector.svg"
        alt="country map"
        className="map_img"
      />
    </div>
  );
};

CountryBox.propTypes = {
  countryName: PropTypes.string.isRequired,
  countryNewCases: PropTypes.number.isRequired,
};

export default CountryBox;
