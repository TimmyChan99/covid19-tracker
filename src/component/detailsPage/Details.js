import React from 'react';
import './details.css';
import { FaVirus } from 'react-icons/fa';
import { useSelector } from 'react-redux';

const Details = () => {
  const countries = useSelector((state) => state.statsReducer);
  const countryName = countries[0].name;
  const selectedCountry = countries ? countries.filter((item) => (item.id === countryName || item.id === 'Total')) : []; // pass default param in the array

  return (
    <div className="country_details">
      <div className="headline">
        <h2>{selectedCountry[0].name}</h2>
        <div className="headline_stat d-flex-column">
          <span>{selectedCountry[0].todayConfirmed}</span>
          <span>Today confirmed</span>
        </div>
        <div className="headline_stat d-flex-column">
          <span>{selectedCountry[1].todayConfirmed}</span>
          <span>Total confirmed</span>
        </div>
        <FaVirus />
      </div>
      <small>ALL STATS</small>
      <div className="today_confirmed box d-flex-column">
        <span className="stat_number">{selectedCountry[0].todayConfirmed}</span>
        <span>today confirmed</span>
      </div>
      <div className="new_death box-odd d-flex-column">
        <span className="stat_number">{selectedCountry[0].todayNewDeaths}</span>
        <span>new death</span>
      </div>
      <div className="new_recovery box-odd d-flex-column">
        <span className="stat_number">{selectedCountry[0].todayNewRecovered}</span>
        <span>new recovery</span>
      </div>
      <div className="today_vs_yesterday_open_cases box d-flex-column">
        <span className="stat_number">{selectedCountry[0].todayVsYesterdayOpenCases.toExponential(2)}</span>
        <span>
          today vs yesterday
          <br />
          {' '}
          open cases
        </span>
      </div>
      <div className="today_vs_yesterday_deaths box d-flex-column">
        <span className="stat_number">{selectedCountry[0].todayVsYesterdayDeaths.toExponential(2)}</span>
        <span>
          today vs yesterday
          <br />
          {' '}
          deaths
        </span>
      </div>
      <div className="today_vs_yesterday_confirmed box-odd d-flex-column">
        <span className="stat_number">{selectedCountry[0].todayVsYesterdayConfirmed.toExponential(2)}</span>
        <span>
          today vs yesterday
          <br />
          {' '}
          confirmed
        </span>
      </div>
      <div className="total_confirmed box-odd d-flex-column">
        <span className="stat_number">{selectedCountry[1].totalTodayConfirmed.toExponential(2)}</span>
        <span>Total confirmed</span>
      </div>
      <div className="total_new_death box d-flex-column">
        <span className="stat_number">{selectedCountry[1].totaltodayNewDeaths}</span>
        <span>Total Death</span>
      </div>
      <div className="total_today_new_recovered box d-flex-column">
        <span className="stat_number">{selectedCountry[1].totalTodayNewRecovered}</span>
        <span>
          total today
          <br />
          {' '}
          new recovered
        </span>
      </div>
      <div className="total_yesterday_deaths box-odd d-flex-column">
        <span className="stat_number">{selectedCountry[1].totalYesterdayDeaths}</span>
        <span>
          total today
          <br />
          {' '}
          new recovered
        </span>
      </div>
      <p>
        Data provided and used for the generation of these products
        comes from the aggregation of different sources,
        each of which have different update times and frequencies.
      </p>
    </div>
  );
};

export default Details;
