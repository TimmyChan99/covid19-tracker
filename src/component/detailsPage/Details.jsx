import React from 'react'
import './details.css'
import { FaVirus } from 'react-icons/fa'
import { useSelector } from 'react-redux'

const Details = () => {
	const selectedCountry = useSelector((state) => state.statsReducer)
	console.log(selectedCountry);
	return (
		<div className="country_details">
			<div className="headline">
       <h2>France</h2>
			 <div className="headline_stat d-flex-column">
				 <span>200</span>
				 <span>Today confirmed</span>
			 </div>
			 <div className="headline_stat d-flex-column">
				 <span>200</span>
				 <span>Total confirmed</span>
			 </div>
			 <FaVirus />
			</div>
			<small>ALL STATS</small>
			<div className="today_confirmed box d-flex-column">
				<span className="stat_number">200</span>
				<span>today confirmed</span>	
			</div>
			<div className="new_death box-odd d-flex-column">
				<span className="stat_number">200</span>
				<span>new death</span>
			</div>
			<div className="new_recovery box-odd d-flex-column">
				<span className="stat_number">200</span>
				<span>new recovery</span>
			</div>
			<div className="today_vs_yesterday_open_cases box d-flex-column">
				<span className="stat_number">200</span>
				<span>today vs yesterday <br /> open cases</span>
			</div>
			<div className="today_vs_yesterday_deaths box d-flex-column">
				<span className="stat_number">200</span>
				<span>today vs yesterday <br /> deaths</span>
			</div>
			<div className="today_vs_yesterday_confirmed box-odd d-flex-column">
				<span className="stat_number">200</span>
				<span>today vs yesterday <br /> confirmed</span>
			</div>
			<div className="total_confirmed box-odd d-flex-column">
				<span className="stat_number">200</span>
				<span>Total confirmed</span>
			</div>
			<div className="total_new_death box d-flex-column">
				<span className="stat_number">200</span>
				<span>Total Death</span>
			</div>
			<div className="total_today_new_recovered box d-flex-column">
				<span className="stat_number">200</span>
				<span>total today <br /> new recovered</span>
			</div>
			<div className="total_yesterday_deaths box-odd d-flex-column">
				<span className="stat_number">200</span>
				<span>total today <br /> new recovered</span>
			</div>
			<p>Data provided and used for the generation of these products
				 comes from the aggregation of different sources,
				 each of which have different update times and frequencies.
			</p>
		</div>
	)
}

export default Details;
