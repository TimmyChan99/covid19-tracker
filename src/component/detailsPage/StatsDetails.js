import React from 'react';
import { useSelector } from 'react-redux';
import CountryBox from './CountryBox';
import { FaVirus } from 'react-icons/fa';

const StatsDetails = () => {
  
  const countries = useSelector((state) => state.statsReducer);
  const countryName = countries[0].name;
  const selectedCountry = countries ? countries.filter((item) => (item.id === countryName || item.id === 'Total')) : defaultData;

  const ids = [1, 2,3,4,5,6,7,8];
  
  return (
    <div className="country_details">
      <div className="headline">
        <h2>COUNTRY</h2>
        <div className="headline_stat d-flex-column">
          <span>601</span>
          <span>Today confirmed</span>
        </div>
        <div className="headline_stat d-flex-column">
          <span>601</span>
          <span>Total confirmed</span>
        </div>
        <FaVirus />
      </div>
      <small>ALL STATS</small>
      <ul className="countries_list">
        { ids.map((item) => (
          <li 
          key={item}
          className="box"
          >
            <CountryBox />
          </li>
        ))}
      </ul>
      
      
    </div>
  );
};

export default StatsDetails;
