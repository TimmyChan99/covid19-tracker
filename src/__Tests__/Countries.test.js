import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from './storeMock';
import CountryBox from '../component/detailsPage/CountryBox';
import StatsDetails from '../component/detailsPage/StatsDetails';
import Details from '../component/detailsPage/Details';

it('Test Region component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CountryBox
          countryName="Country"
          countryNewCases={152}
        />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});


it('Test Region component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Details />
      </BrowserRouter>
    </Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
