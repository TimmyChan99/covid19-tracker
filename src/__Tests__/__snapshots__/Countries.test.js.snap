// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Region component 1`] = `
<div
  className="country d-flex-column"
>
  <a
    className="arrow"
    href="/covid19-tracker/countries-list/Country/regions/"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      id="Country"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        fillRule="evenodd"
      />
    </svg>
  </a>
  <div
    className="info d-flex-column"
  >
    <span
      className="country_name"
    >
      Country
    </span>
    <span
      className="stat_number"
    >
      152
       
      cases
    </span>
  </div>
  <img
    alt="country map"
    className="map_img"
    src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/ad/vector.svg"
  />
</div>
`;

exports[`Test Region component 2`] = `
<div>
  <div
    className="message_err"
  >
    <h3>
      No Statistics Found
    </h3>
    <span>
      Try again By changing the date
    </span>
  </div>
</div>
`;
